### PR TITLE
Disable unstable JS _show ETag test

### LINKED ISSUE #559: 304 after GET-PUT-GET cycle on a _show
## Expected & Current Behaviour
1. Why is a GET-PUT-GET cycle returning a 304 on the second GET with headers "if-none-match" etag for a `_show`?

2. Misleading error message should be cleaned up

makefile output:
```
test/javascript/tests/show_documents.js                        
    Error: expected '304', got '304'
Trace back (most recent call first):
    
  52: test/javascript/test_setup.js
      T(false,"expected '304', got '304'","changed ddoc")
 326: test/javascript/couch_test_runner.js
      TNotEquals(304,304,"changed ddoc")
 296: test/javascript/tests/show_documents.js
      ()
  37: test/javascript/cli_runner.js
      runTest()
  48: test/javascript/cli_runner.js
      
[31mfail
```

couch.log:
```
[notice] 2017-05-31T23:29:00.503101Z node1@127.0.0.1 <0.16311.3> cea9c20ff0 127.0.0.1:15984 127.0.0.1 undefined GET /test_suite_db_xofwxqrc/_design/template/_show/just-name/78db53e02b06ddd067440cd0dd42f84c 304 ok 1
[notice] 2017-05-31T23:29:00.542915Z node1@127.0.0.1 <0.16311.3> 5cf1142bed 127.0.0.1:15984 127.0.0.1 undefined PUT /test_suite_db_xofwxqrc/_design%2Ftemplate 201 ok 40
[notice] 2017-05-31T23:29:00.545495Z node1@127.0.0.1 <0.16311.3> 58075015b3 127.0.0.1:15984 127.0.0.1 undefined GET /test_suite_db_xofwxqrc/_design/template/_show/just-name/78db53e02b06ddd067440cd0dd42f84c 304 ok 2
```

## Your Environment
Jenkins, Erlang 18.3

### PR DESCRIPTION
**Pull request by wohali**
The full discussion is in #559, but here is a summary.

Through instrumentation of ddoc_cache and ets_lru I suspect that this is
caused by cache eviction happening after the second GET. While I didn't
instrument exactly where the GET occurs it's clear that it's fairly
late, certainly after the PUT 201 is returned, and likely after the
subsequent GET actually reads from ddoc_cache.

After applying a change to allow me to completely disable the ddoc_cache
(-ddoc_cache max_objects 0 in vm.args) I ran the test on a loop
overnight, and the test never failed (>1000 executions). Previously the
test would fail every 20-30 executions.

TL;DR: we can't guarantee immediate ddoc_cache eviction on a ddoc
update, even for a single .couch file on a single node. (For obvious
reasons we definitely can't guarantee this in a cluster configuration.)

I will document this as a backwards compatibility change in 2.0 and
forward with a separate checkin to couchdb-documentation.

Thanks to @rnewson @janl and @davisp for helping track this one down!

This checkin also includes an improvement to the output when a JS test
fails a notEquals assertion.

Closes #559

### CODE REVIEW

**Comment by wohali** [comment]
Pre-approved by @janl and @davisp in #559. Rebasing and merging now.
